src/core/server/metrics/collectors/server.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { ResponseObject, Server as HapiServer } from 'hapi';
+import { OpsServerMetrics, MetricsCollector } from './types';
+
+interface ServerResponseTime {
+  count: number;
+  total: number;
+  max: number;
+}
+
+export class ServerMetricsCollector implements MetricsCollector<OpsServerMetrics> {
+  private readonly requests: OpsServerMetrics['requests'] = {
+    disconnects: 0,
+    total: 0,
+    statusCodes: {},
+  };
+  private readonly responseTimes: ServerResponseTime = {
+    count: 0,
+    total: 0,
+    max: 0,
+  };
+
+  constructor(private readonly server: HapiServer) {
```

Comment:
What if we prevent `hapi server` leakage and invert dependencies? The HTTP server could implement the `collect` interface instead.

Reply:
We are inside core and no `hapi` reference is leaking outside of it, so I would say it's alright. But I can move the `server` collector inside `http` and expose it from the internal http contract is you think this is better / more futur proof.

Reply:
We are inside core and no `hapi` reference is leaking outside of it, so I would say it's alright. But I can move the `server` collector inside `http` and expose it from the internal http contract is you think this is better / more futur proof.

Reply:
> We are inside core and no hapi reference is leaking outside of it, so I would say it's alright. But I can move the server collector inside http and expose it from the internal http contract is you think this is better / more futur proof.

It's up to you, but I'd prefer to have the one place to update if we decide to get rid of hapi one day.